### PR TITLE
 Added Backspace String Compare problem in DSA

### PR DESCRIPTION
**Pull request by rahulch-1**

### CODE REVIEW

**Comment by iamwatchdogs** [comment]
@rahulch-1 Please don't create another PR when you make your changes. You can commit your changes within the same PR, all you have to do is commit on the same branch that you're making the PR.

**Comment by iamwatchdogs** [comment]
@rahulch-1 We have fixed your code for you to pass the linting checks.

File: Solved-Problems/Backspace_String_Compare/Main.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import java.util.Stack;
```
Comment (iamwatchdogs) [inline Solved-Problems/Backspace_String_Compare/Main.java:1]: @rahulch-1 Please provide the problem-related details within your program file as mentioned within the rules and guidelines in [CONTRIBUTING.md](../../blob/main/CONTRIBUTING.md#instructions-and-guidelines).

```suggestion
// Problem Name: <problem-name>
// Approach:     <data-structure> (or) <algorithm>
// Diffculty:    <mentioned-diffculty>
// Link:         <link-to-the-problem>

import java.util.Stack;
```

**Review by iamwatchdogs — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: Solved-Problems/Backspace_String_Compare/Main.java [hunk 2]
```diff
@@ -1,3 +1,12 @@
+
+//Problem Name: 844.Backspace String Compare
+
+//Approach: Using Stacks
+
+//Difficulty: Easy
+
+//Link: https://leetcode.com/problems/backspace-string-compare/
```
Comment (iamwatchdogs) [inline Solved-Problems/Backspace_String_Compare/Main.java:8]: I believe it's a simple misunderstanding, there's no gap between the comments. Please remove those unnecessary new lines.

```suggestion
//Problem Name: 844.Backspace String Compare
//Approach: Using Stacks
//Difficulty: Easy
//Link: https://leetcode.com/problems/backspace-string-compare/
```

**Review by iamwatchdogs — CHANGES_REQUESTED** [review]
Please make these final changes, so that we can merge your pull request.

**Comment by rahulch-1** [comment]
Done the changes @iamwatchdogs

**Review by iamwatchdogs — APPROVED** [review]
The requested changes have been made. Review and Approved by @iamwatchdogs.

**Comment by iamwatchdogs** [comment]
Thank you for contributing @rahulch-1, make sure to check your contributions on [GitHub Pages](https://grow-with-open-source.github.io/DSA/).

**Pull request merged**